drivers/serial/uart_ite_it8xxx2.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2021 ITE Corporation. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8xxx2_uart
+
+#include <device.h>
+#include <drivers/gpio.h>
+#include <drivers/uart.h>
+#include <kernel.h>
+#include <pm/device.h>
+#include <soc.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(uart_ite_it8xxx2, LOG_LEVEL_ERR);
+
+struct uart_it8xxx2_config {
+	uint8_t port;
+	/* GPIO cells */
+	struct gpio_dt_spec gpio_wui;
+};
+
+enum uart_port_num {
+	UART1 = 1,
+	UART2,
+};
+
+#ifdef CONFIG_PM_DEVICE
+__weak void uart1_wui_isr_late(void) { }
+__weak void uart2_wui_isr_late(void) { }
+
+void uart1_wui_isr(const struct device *gpio, struct gpio_callback *cb,
+		   uint32_t pins)
+{
+	/* Disable interrupts on UART1 RX pin to avoid repeated interrupts. */
+	(void)gpio_pin_interrupt_configure(gpio, (find_msb_set(pins) - 1),
+					   GPIO_INT_DISABLE);
+
+	if (uart1_wui_isr_late) {
+		uart1_wui_isr_late();
+	}
+}
+
+void uart2_wui_isr(const struct device *gpio, struct gpio_callback *cb,
+		   uint32_t pins)
+{
+	/* Disable interrupts on UART2 RX pin to avoid repeated interrupts. */
+	(void)gpio_pin_interrupt_configure(gpio, (find_msb_set(pins) - 1),
+					   GPIO_INT_DISABLE);
+
+	if (uart2_wui_isr_late) {
+		uart2_wui_isr_late();
+	}
+}
+
+static inline int uart_it8xxx2_pm_action(const struct device *dev,
+					 enum pm_device_action action)
+{
+	const struct uart_it8xxx2_config *const config = dev->config;
+	int ret = 0;
+
+	switch (action) {
+	/* Next device power state is in active. */
+	case PM_DEVICE_ACTION_RESUME:
+		/* Nothing to do. */
+		break;
+	/* Next device power state is deep doze mode */
+	case PM_DEVICE_ACTION_SUSPEND:
+		/* Enable UART WUI */
+		ret = gpio_pin_interrupt_configure(config->gpio_wui.port,
```

Comment:
please, use `gpio_pin_interrupt_dt_configure`

Reply:
have you considered using https://docs.zephyrproject.org/latest/doxygen/html/group__subsys__pm__sys.html#ga066945118b8f287ee56aacf41b677a78 ?

Reply:
Thank you for your suggestion, I will make changes in the future PR. 
But I want to know the difference between doing configuration here and doing pm_notifier_register(). Is the registration in this callback function?

Reply:
> please, use `gpio_pin_interrupt_dt_configure`

Done.